airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py

```diff
@@ -37,21 +39,104 @@ class GoogleAnalyticsSingerSource(BaseSingerSource):
     api_error = Exception
+    _custom_reports_schema = {
+        "$schema": "http://json-schema.org/schema#",
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "name": {"type": "string"},
+                "dimensions": {
+                    "type": "array",
+                    "title": "dimensions",
+                    "items": {
+                        "type": "string",
+                        "enum": [
+                            "ga:browser",
+                            "ga:city",
+                            "ga:continent",
+                            "ga:country",
+                            "ga:date",
+                            "ga:deviceCategory",
+                            "ga:hostname",
+                            "ga:medium",
+                            "ga:metro",
+                            "ga:operatingSystem",
+                            "ga:pagePath",
+                            "ga:region",
+                            "ga:socialNetwork",
+                            "ga:source",
+                            "ga:subContinent",
+                        ],
+                    },
+                    "maxItems": 7,
+                    "uniqueItems": True,
+                },
+                "metrics": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "enum": [
+                            "ga:14dayUsers",
+                            "ga:1dayUsers",
+                            "ga:28dayUsers",
+                            "ga:30dayUsers",
+                            "ga:7dayUsers",
+                            "ga:avgSessionDuration",
+                            "ga:avgTimeOnPage",
+                            "ga:bounceRate",
+                            "ga:entranceRate",
+                            "ga:entrances",
+                            "ga:exitRate",
+                            "ga:exits",
+                            "ga:newUsers",
+                            "ga:pageviews",
+                            "ga:pageviewsPerSession",
+                            "ga:sessions",
+                            "ga:sessionsPerUser",
+                            "ga:uniquePageviews",
+                            "ga:users",
+                        ],
+                    },
+                    "maxItems": 10,
+                    "uniqueItems": True,
+                },
+            },
+        },
+    }
 
     # can be overridden to change an input config
     def configure(self, raw_config: json, temp_dir: str) -> json:
         credentials = os.path.join(temp_dir, "credentials.json")
         with open(credentials, "w") as fh:
             fh.write(raw_config["credentials_json"])
         raw_config["key_file_location"] = credentials
+        if raw_config["custom_reports"].strip() and json.loads(raw_config["custom_reports"]):
```

Comment:
shouldn't this be `if "custom_reports" in raw_config and raw_config["custom_reports"].strip()...` ? Basically this would throw a key error if custom_reports is not in raw_config, right?

Reply:
Yes, I missed this moment, I have already fixed it, though in a different way.